packages/api/src/services/erc20-state.ts

```diff
@@ -0,0 +1,40 @@
+import { clients } from "@uma/sdk";
+import { Libs, Json } from "..";
+import { asyncValues } from "../libs/utils";
+type Config = Json;
+export default function (config: Config, libs: Libs) {
+  const { provider, erc20s, collateralAddresses, syntheticAddresses } = libs;
```

Comment:
I'm a little confused at the naming... why are the addresses to be tracked passed in as a lib? Seems like an odd name?

Also, as a secondary note: the Libs type seems to be quite expansive -- is libs a global object that's expected to be static and shared? Or should the requirements for each of these methods be down-specified to only the variables they actually use/need? Just seems odd to couple all the signatures of somewhat unrelated functions.

Reply:
good point, the naming sucks here, the 'libs' really represent state shared across the app and between services. Some service may update state, while another might query state.  For simplicity the entire state is composed into a single object defined by the `Libs` type. The state can be tables, sets, objects, whatever is the best way to represent the data. 

I am very open to renaming this, `State` seems too general, but maybe it could work, maybe `SharedState`, but i dont know

Reply:
im going to rename this to `AppState` for now